test/visualizer-data-tree.js

```diff
@@ -0,0 +1,43 @@
+const test = require('tap').test
+const DataTree = require('../visualizer/data-tree.js')
+
+test('visualizer - data tree - sort frames by code area and stack top value', (t) => {
+  const rootNode = {
+    name: 'all stacks',
+    value: 2,
+    onStackTop: { base: 2 },
+    // Core totals 200
+    // Fastify totals 260
+    // → fastify should be first
+    children: [
+      { type: 'fastify', value: 100, onStackTop: { base: 100 }, children: [] },
+      { type: 'core', value: 200, onStackTop: { base: 200 }, children: [] },
+      { type: 'fastify', value: 20, onStackTop: { base: 20 }, children: [] },
+      { type: 'fastify', value: 40, onStackTop: { base: 40 }, children: [] },
+      { type: 'fastify', value: 100, onStackTop: { base: 100 }, children: [] }
+    ]
+  }
+
+  const tree = new DataTree({
+    merged: rootNode,
+    unmerged: rootNode,
+    appName: 'test',
+    pathSeparator: '/'
+  })
+
+  tree.update()
+  const sorted = tree.activeNodes().slice()
+    .sort(tree.getFilteredStackSorter())
+
+  // Use match() rather than same(), because DataTree adds some properties
+  // like `onStackTop.asViewed` that are not relevant for this test
+  t.match(sorted, [
```

Comment:
Good test! One thought to maybe make it a bit less brittle: we don't need to test against the complete sorted object (which would then all need updating any time a property is added or changed on the data model), we could slim it down like:

```
function slimmed ({ type, value }) {
  return { type, value }
}

t.match(sorted.map(slimmed), [
  { type: someType, value: 100 },
  { type: someType, value: 50 },
  ... etc
]
```

...testing only whichever properties we need to test to ensure the order is right.

Reply:
good idea. all the `onStackTop` and `children` stuff must be a leftover from using `t.same()` before changing to `t.match()`. `t.match()` already ignores all additional properties so we can just remove the irrelevant stuff, don't even need the `slimmed` function! :smile_cat: